File: src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

class MyNode(Node):

    def __init__(self):
        super().__init__("py_test")
        self.counter_ = 1
        self.create_timer(5, self.timer_callback)
        self.get_logger().info("Hello World")

    def timer_callback(self): 
        self.get_logger().info("Hello" + str(self.counter_))
        self.counter_ += 1




def main(args=None):
    # just testing
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"
#include <format>

class MyNode : public rclcpp::Node 
{
public:
    MyNode() : Node("cpp_test"), counter(0)
    {
        RCLCPP_INFO(this->get_logger(), "Hello World");
        timer = this->create_wall_timer(std::chrono::seconds(2), 
        
        [&]() { timerCallback(); }
        //std::bind(&MyNode::timerCallback, this)
    
    );
    }
private:
    void timerCallback()
    {
        RCLCPP_INFO(this->get_logger(), std::format("Hello {}", counter++).c_str());
        
    }

    rclcpp::TimerBase::SharedPtr timer;
    int counter;

};

int main(int argc, char ** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}